### PR TITLE
Add ListOptions to ListWatched

### PR DESCRIPTION
**Pull request by sona-tar**
Supoprt #238

### CODE REVIEW

File: github/activity_watching.go [hunk 1]
```diff
@@ -50,13 +50,14 @@ func (s *ActivityService) ListWatchers(owner, repo string, opt *ListOptions) ([]
...
 // GitHub API Docs: https://developer.github.com/v3/activity/watching/#list-repositories-being-watched
-func (s *ActivityService) ListWatched(user string) ([]Repository, *Response, error) {
+func (s *ActivityService) ListWatched(user string, opt *ListOptions) ([]Repository, *Response, error) {
 	var u string
 	if user != "" {
 		u = fmt.Sprintf("users/%v/subscriptions", user)
 	} else {
 		u = "user/subscriptions"
 	}
+	u, err := addOptions(u, opt)
```
Comment (parkr) [inline github/activity_watching.go]: Does this handle the `nil` case for `opt`? Could the tests be amended to ensure this?

**Comment by sona-tar** [comment]
I fixed it!

**Comment by sona-tar** [comment]
@willnorris  ping.
 Can we get feedback on this PR?

File: github/activity_watching.go (hunk 1 above)
Comment (willnorris) [inline github/activity_watching.go]: yes, `addOptions` handles nil values.  I see now that we don't any tests for `addOptions`, but that would be the place to test this, not in every method that happens to call it (which is hundreds).

**Comment by willnorris** [comment]
made above changes I mentioned and merged as c57bb02.  Thanks!

**Pull request closed without merging**